tests/api_client/test_api_client.py

```diff
@@ -0,0 +1,309 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Unit tests for API client
+"""
+
+import pytest
+import json
+from strawberryfields import api_client
+from strawberryfields.api_client import (
+    requests,
+    Job,
+    ResourceManager,
+    ObjectAlreadyCreatedException,
+    MethodNotSupportedException,
+)
+
+from unittest.mock import MagicMock
```

Comment:
Aren't `MagicMock` and `monkeypatch` two tools for doing the same thing? How come we're using both?

Reply:
Actually had that question as well

Reply:
monkeypatch allows you to patch an attribute on an object (including one that you imported, for example.) MagicMock is itself a mock object that you can patch however way you want. I'm using Pytest's monkeypatch here because it's easy to use to patch one or a few attributes of an object, whereas MagicMock is used as mock objects (rather than creating mock objects from scratch). Thoughts?

Reply:
Makes sense!